### PR TITLE
feat: update the Places Demo with a map to show the current place

### PR DESCRIPTION
**Pull request by johnshea**

### CODE REVIEW

**Review by dkhawk — CHANGES_REQUESTED** [review]

File: solution/app/build.gradle [hunk 1]
```diff
@@ -48,4 +48,17 @@ dependencies {
...
+}
+
+secrets {
+    // To add your Maps API key to this project:
+    // 1. Open the root project's local.properties file
+    // 2. Add this line, where YOUR_API_KEY is your API key:
+    //        MAPS_API_KEY=YOUR_API_KEY
+    propertiesFileName = "secrets.properties"
+    defaultPropertiesFileName 'local.defaults.properties'
+
```
[the remark this answers is not in the document]
Comment (johnshea) [reply solution/app/build.gradle]: Removed as requested.

**Review by dkhawk — APPROVED** [review]

**Review by wangela — APPROVED** [review]

**Review by dkhawk — APPROVED** [review]

File: solution/.gitignore [hunk 2]
```diff
@@ -81,3 +81,4 @@ lint/generated/
 lint/outputs/
 lint/tmp/
 # lint/reports/
+/secrets.properties
```
Comment (wangela) [inline solution/.gitignore:84]: I think this leading slash may not work

File: solution/app/build.gradle [hunk 3]
```diff
@@ -24,28 +25,38 @@ android {
...
+    implementation 'androidx.lifecycle:lifecycle-runtime-ktx:2.7.0'
     testImplementation 'junit:junit:4.13.2'
-    androidTestImplementation 'androidx.test.ext:junit:1.1.3'
-    androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
+    androidTestImplementation 'androidx.test.ext:junit:1.1.5'
+    androidTestImplementation 'androidx.test.espresso:espresso-core:3.5.1'
 
     // Include Places SDK for Android
-    implementation 'com.google.android.libraries.places:places:2.6.0'
+    implementation 'com.google.android.libraries.places:places:3.3.0'
```
Comment (wangela) [inline solution/app/build.gradle:49]: We are now on 3.4.0. Should be an easy upgrade (no code changes)

File: solution/app/build.gradle [hunk 4]
```diff
@@ -24,28 +25,38 @@ android {
...
     // Places SDK for Android KTX Library
-    implementation 'com.google.maps.android:places-ktx:2.0.0'
+    implementation 'com.google.maps.android:places-ktx:3.1.1'
+    implementation 'com.google.android.gms:play-services-maps:18.2.0'
+}
+
+secrets {
+    // To add your Maps API key to this project:
+    // 1. Open the root project's local.properties file
```
Comment (wangela) [inline solution/app/build.gradle:57]: Match the code snippet in the codelab fragment

File: solution/app/src/main/AndroidManifest.xml [hunk 5]
```diff
@@ -29,6 +28,16 @@
         android:roundIcon="@mipmap/ic_launcher_round"
         android:supportsRtl="true"
         android:theme="@style/Theme.PlacesDemo">
+        <meta-data
+            android:name="com.google.android.gms.version"
+            android:value="@integer/google_play_services_version" />
+        <!--
+             The API key for Google Maps-based APIs.
```
Comment (wangela) [inline solution/app/src/main/AndroidManifest.xml:35]: Maps SDK for Android

(this key in the Manifest is not used for the Places SDK)

File: starter/.gitignore [hunk 6]
```diff
@@ -81,3 +81,4 @@ lint/generated/
 lint/outputs/
 lint/tmp/
 # lint/reports/
+/secrets.properties
```
Comment (wangela) [inline starter/.gitignore:84]: no leading slash

**Review by wangela — CHANGES_REQUESTED** [review]

**Review by wangela — CHANGES_REQUESTED** [review]
Please merge in changes from #14

File: solution/.gitignore (hunk 2 above)
Comment (johnshea) [reply solution/.gitignore]: Remove slash.

File: solution/app/build.gradle (hunk 3 above)
Comment (johnshea) [reply solution/app/build.gradle]: Version updated.

File: solution/app/build.gradle (hunk 4 above)
Comment (johnshea) [reply solution/app/build.gradle]: Updated so it matches.

File: solution/app/src/main/AndroidManifest.xml (hunk 5 above)
Comment (johnshea) [reply solution/app/src/main/AndroidManifest.xml]: Changed as requested.

File: starter/.gitignore (hunk 6 above)
Comment (johnshea) [reply starter/.gitignore]: Removed slash.

**Comment by johnshea** [comment]
> Please merge in changes from #14

These changes are already on this branch from https://github.com/googlemaps-samples/codelab-places-101-android-kotlin/pull/13/commits/63775da3be75608017a4e7ed4fc758b458795667.

No additional work required.

File: solution/app/src/main/java/com/google/codelabs/maps/placesdemo/StringUtil.kt [hunk 7]
```diff
@@ -16,137 +16,68 @@
...
-        builder.append("Fetch Place Result:").append(RESULT_SEPARATOR)
-        if (raw) {
-            builder.append(response.place)
-        } else {
-            builder.append(stringify(response.place))
-        }
-        return builder.toString()
-    }
+fun FindCurrentPlaceResponse.prettyPrint(): String {
+    val raw = false
```
Comment (dkhawk) [inline solution/app/src/main/java/com/google/codelabs/maps/placesdemo/StringUtil.kt:44]: Same here.  Drop the raw val

Comment (johnshea) [reply solution/app/src/main/java/com/google/codelabs/maps/placesdemo/StringUtil.kt]: Done.

File: solution/app/src/main/java/com/google/codelabs/maps/placesdemo/StringUtil.kt [hunk 8]
```diff
@@ -16,137 +16,68 @@
...
-        textView.text = """
-            $prefix
+internal const val FIELD_SEPARATOR = "\n\t"
+internal const val RESULT_SEPARATOR = "\n---\n\t"
 
-            ${textView.text}
-            """.trimIndent()
-    }
+fun FetchPlaceResponse.prettyPrint(): String {
+    val raw = false
```
[the remark this answers is not in the document]
Comment (johnshea) [reply solution/app/src/main/java/com/google/codelabs/maps/placesdemo/StringUtil.kt]: Done.

**Review by dkhawk — APPROVED** [review]

**Review by wangela — APPROVED** [review]

**Pull request merged**